pkg/gcp/machine_controller_test.go

```diff
@@ -120,6 +120,10 @@ var _ = Describe("#MachineController", func() {
 		"userData":                []byte("dummy-data"),
 		api.GCPServiceAccountJSON: []byte("{\"type\":\"service_account\",\"project_id\":\"sap-se-gcp-scp-k8s-dev\"}"),
 	}
+	gcpProviderSecretWithCredentialsConfig := map[string][]byte{
+		"userData":               []byte("dummy-data"),
+		api.GCPCredentialsConfig: []byte("{\"type\":\"service_account\",\"project_id\":\"sap-se-gcp-scp-k8s-dev\"}"),
```

Comment:
I see you are following the already established code style here, but you might want to:
1. do not use the constants in tests, this way if the constant value is changed, the test will fail and make the change apparent
2. you can avoid escaping the double quotes by using back ticks 

 ```suggestion
		"credentialsConfig": []byte(`{"type":"service_account","project_id":"sap-se-gcp-scp-k8s-dev"}`),
```